components/script/microtask.rs

```diff
@@ -71,6 +85,15 @@ impl MicrotaskQueue {
                             let _ = job.callback.Call_(&*target, ExceptionHandling::Report);
                         }
                     }
+                    Microtask::Mutation(ref job) => {
+                        if let Some(target) = target_provider(job.pipeline) {
```

Comment:
Here we should be invoking `Microtask::notifyMutationObservers`, since this microtask should be implementing the task queued in step 3 of https://dom.spec.whatwg.org/#queue-a-mutation-observer-compound-microtask.